devito/ir/clusters/algorithms.py

```diff
@@ -33,6 +33,7 @@ def clusterize(exprs, **kwargs):
 
     # Setup the IterationSpaces based on data dependence analysis
     clusters = impose_total_ordering(clusters)
+    clusters = petsc_lift(clusters)
```

Comment:
why not a pass in `_specialize_clusters`? it's conceptually wrong that this happens here, especially before `Schedule`

Reply:
or inside `operator._lower_clusters` right after `clusterize`